maven-stax-xinclude/src/main/java/org/apache/maven/stax/xpointer/ElementPointerPart.java

```diff
@@ -0,0 +1,245 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.maven.stax.xpointer;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * ElementPointerPart is a class which represents the element() scheme for the XPointer Framework.
+ * The specification is defined at <a href="http://www.w3.org/TR/xptr-element/">http://www.w3.org/TR/xptr-element/</a>
+ * <p>
+ * This class is immutable.
+ * <p>
+ * This class is based upon a class of the same name in the Apache Woden.
+ */
+public class ElementPointerPart implements PointerPart {
```

Comment:
This PR exposes a lot of new public API that's going to be hard and expensive to support. Is there anyway to make less of it public? E.g. by using a single package?

Reply:
FTR, I do really question the fact that a class being public is supposed to be supported when consumed by other projects.   Especially with Maven 4, the API will be defined by everything in under the `org.apache.maven.api` and anything not under this package will not be part of the API.

Anyway, I'll refactor into a single package with less public classes...